File: app/src/main/java/com/example/plantonic/Adapter/TrackStatusAdapter.java
package com.example.plantonic.Adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.plantonic.R;
import com.example.plantonic.ShipmentResponse;

import java.util.List;

public class TrackStatusAdapter extends RecyclerView.Adapter<TrackStatusAdapter.ViewHolder> {
    private final List<ShipmentResponse.ScanDetail> scanDetailList;
    private final Context context;

    public TrackStatusAdapter(Context context, List<ShipmentResponse.ScanDetail> scanDetailList) {
        this.context = context;
        this.scanDetailList = scanDetailList;
    }

    @NonNull
    @Override
    public TrackStatusAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_shipment_details, parent, false);
        return new ViewHolder(view);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ShipmentResponse.ScanDetail scanDetail = scanDetailList.get(position);
        holder.statusDate.setText("Scan Date: " + scanDetail.getScanDate());
        holder.statusTitle.setText("Scan: " + scanDetail.getScan());
        holder.statusTime.setText("Scan Time: " + scanDetail.getScanTime());
    }

    @Override
    public int getItemCount() {
        return scanDetailList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView statusTitle, statusDate, statusTime;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            statusTitle = itemView.findViewById(R.id.statusTitle);
            statusDate = itemView.findViewById(R.id.statusDate);
            statusTime = itemView.findViewById(R.id.statusTime);
        }
    }
}

File: app/src/main/java/co/in/plantonic/Adapter/TrackStatusAdapter.kt
package co.`in`.plantonic.Adapter

import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import co.`in`.plantonic.R
import android.content.Context
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import co.`in`.plantonic.retrofit.models.track.ScanDetail

class TrackStatusAdapter(
    private val context: Context,
    private val scanDetailList: List<ScanDetail>
) : RecyclerView.Adapter<TrackStatusAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view =
            LayoutInflater.from(context).inflate(R.layout.item_shipment_details, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val scanDetail = scanDetailList[position]
        holder.trackDate.text = scanDetail.ScanDate
        holder.trackTitle.text = scanDetail.Scan
        holder.trackLocation.text = scanDetail.ScannedLocation
        holder.trackTime.text = scanDetail.ScanTime
        if (position == scanDetailList.size - 1){
            holder.trackLowerLine.visibility = View.GONE
        }else{
            holder.trackLowerLine.visibility = View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return scanDetailList.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var trackTitle: TextView
        var trackDate: TextView
        var trackTime: TextView
        var trackLocation: TextView
        var trackLowerLine: ImageView

        init {
            trackTitle = itemView.findViewById<TextView>(R.id.trackTitle)
            trackDate = itemView.findViewById<TextView>(R.id.trackDate)
            trackTime = itemView.findViewById<TextView>(R.id.trackTime)
            trackLocation = itemView.findViewById<TextView>(R.id.trackLocation)
            trackLowerLine = itemView.findViewById<ImageView>(R.id.trackLowerLine)
        }
    }
}